### ISSUE TITLE
Pan Gestures conflict with ScrollView?

### ISSUE THREAD

**Issue by rgoldiez** [root-two/react-native-drawer#52]
Still trying to understand how `react-native-drawer` works (maybe react native, more generally).  In the below example, the Dashboard component has a ScrollView to allow panning in the vertical direction.  This works great, except in the area where `react-native-drawer` is waiting for the panGesture (the panOpenMask property).  In this area, only a horizontal swipe is recognized and it opens/closes the drawer.  

Is this the expected behavior, or can the vertical gestures be left to the ScrollView?

```
   render: function() {
      return (
         <View style={styles.mainContainer}>
            <Drawer
               ref='leftDrawer'
               content={<LeftDrawer />}
               type='static'
               side='left'
               openDrawerOffset={75}
               onOpen={this.openLeftDrawer}
               onClose={this.closeDrawers}
               styles={{main: {shadowColor: '#000000', shadowOpacity: 0.4, shadowRadius: 3}}}
               tweenHandler={Drawer.tweenPresets.parallax}
               tweenDuration={100}
               >
               <View style={styles.container}>
                  <NavBar toggleLeftDrawer={this.toggleLeftDrawer}/>
                  <Dashboard />
               </View>
            </Drawer>
         </View>
```

**Comment by rt2zz** [comment]
The way react-native gesture negotiation works is that children have first dibs to handle a pan. This means that your scroll view should work and take precedence over the drawer pan.

The exception to that is if `captureGestures={true}` which inverts the negotiation flow and gives first dibs to the parent.

Depending on the details of your implementation there may be a few ways to "resolve" this issue. If you need to have `captureGestures={true}`, the simplest solution is typically to make your `panOpenMask={5}` which means only 5 pixels worth of the edge are valid to start a drawer open pan. In normal use no user starts a vertical pan that close to the edge, thus resolving the issue.

Let me know if that works

**Comment by rgoldiez** [comment]
Zack - Thanks for the feedback.  I added `panOpenMask` and anything > 1 is giving the entire screen to `react-native-drawer`.  If I make it 0.5, the left half of the screen is available to open the drawer (with a swipe from left to right (left hand drawer)), but I can't vertically scroll in that section.  The right half of the screen is available for vertical scrolling (w/ the mask set to 0.5).  Am I doing something obviously wrong?  Using 1.90 of `react-native-drawer`

Thanks for the help.

**Comment by rt2zz** [comment]
Sounds like a bug, ill check it out

**Comment by rt2zz** [comment]
added pixel based masking in v1.9.1

Let me know if that solves your issue.

**Comment by rgoldiez** [comment]
Just tried 1.9.1 .... pixel based masking does work, but I get the same behavior.  I set the mask to 5 and you have to start at the edge to swipe open the drawer (not the behavior I'd like) but the vertical scrolling works (except for those first 5 pixels).  I'm looking for Slack type behavior...... left right opens drawers.  Up down scrolls the ScrollView.

**Comment by rt2zz** [comment]
ah ok. Right now react-native-drawer only implements `onStartShouldSetPanResponder`.

To fully achieve slack (iOS) style you are going to need to implement [`onMoveShouldSetPanResponder`](http://facebook.github.io/react-native/docs/panresponder.html) and return true to what you consider to be "horizontal" moves. It will take some tinkering to get right, but should not be paricularly difficult.

Basically you would need to implement a method similar to https://github.com/rt2zz/react-native-drawer/blob/master/index.js#L246 but which only returns true if a horizontal gesture is detected. I think you will also need to disable `onStartShouldSetPanResponder` to allow the negotiation to happen.

If you get it working I will gladly accept a PR!

**Comment by rgoldiez** [comment]
Created a PR for your review.  Thanks for all your help with this.
